Makefile

```diff
@@ -1,56 +1,56 @@
-default: prepare-build
-default: build/holo build/holo-files
-default: build/man/holorc.5 build/man/holo-plugin-interface.7 build/man/holo-test.7 build/man/holo.8 build/man/holo-files.8
+BUILD_TARGETS += $(patsubst cmd/%,gopath/bin/%, $(wildcard cmd/*))         # build all binaries
+BUILD_TARGETS += $(patsubst doc/%.pod,gopath/man/%, $(wildcard doc/*.pod)) # build all manpages
+
+default: $(BUILD_TARGETS)
 
 VERSION := $(shell ./util/find_version.sh)
 
+GO            := env GOPATH=$(CURDIR)/gopath go
 GO_BUILDFLAGS :=
 GO_LDFLAGS    := -s -w
 
-prepare-build:
-	@mkdir -p build/man
-build/holo: src/holo/main.go src/holo/*/*.go
-	go build $(GO_BUILDFLAGS) --ldflags "$(GO_LDFLAGS) -X main.version=$(VERSION)" -o $@ $<
-build/holo-files: src/holo-files/main.go src/holo-files/*/*.go
-	go build $(GO_BUILDFLAGS) --ldflags "$(GO_LDFLAGS)" -o $@ $<
+gopath/bin/holo: cmd/holo/main.go pkg/holo/*.go
+	$(GO) install $(GO_BUILDFLAGS) --ldflags "$(GO_LDFLAGS) -X github.com/holocm/holo/cmd/holo.version=$(VERSION)" github.com/holocm/holo/cmd/holo
+gopath/bin/holo-files: cmd/holo-files/main.go pkg/holo-files/*/*.go
+	$(GO) install $(GO_BUILDFLAGS) --ldflags "$(GO_LDFLAGS)" github.com/holocm/holo/cmd/holo-files
```

Comment:
It's not safe to run `go install` on the same workspace in parallel; they will stomp on each other; meaning that if parallel Make is enabled (as it is for users like me who stick `export MAKEFLAGS="-j $(nproc)"` in their `~/.profile`), then this Makefile is broken.

Instead, all of the needed packages should be names as arguments to a single instance of `go install`; which includes its own logic for when to rebuild.

In the Makefile in the coverage-testing PR, rather than bothering to try to enumerate all of the `.go` files for the Makefile (duplicating `go install`'s logic), I force it to run `go install` every time, unconditionally, and let `go install` rather than `make` decide when there is no work to be done.